contracts/PoolManager.sol

```diff
@@ -388,6 +374,13 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
         emit HookFeeUpdated(id, newHookFees);
     }
 
+    function updateDynamicSwapFee(PoolKey memory key) external {
+        uint24 newDynamicSwapFee = _fetchDynamicSwapFee(key);
```

Comment:
if this is not a dynamic swap fee pool, this will return 0 and set the fee to 0 i think :scream:

Reply:
think `_fetchDynamicSwapFee` should revert if not dynamic

Reply:
or maybe just call it `_fetchFee` which fetches from the key if static or from `getFee` if dynamic

Reply:
Very good catchhh thanks. Moved it up a level